tests/testStreamOpen.cpp

```diff
@@ -85,14 +85,39 @@ TEST_F(StreamOpen, OutputForOpenSLESPerformanceModeShouldBeNone){
 
 TEST_F(StreamOpen, InputForOpenSLESPerformanceModeShouldBeNone){
     // We will not get a LowLatency stream if we request 16000 Hz.
+    mBuilder.setAudioApi(AudioApi::OpenSLES);
     mBuilder.setSampleRate(16000);
     mBuilder.setPerformanceMode(PerformanceMode::LowLatency);
     mBuilder.setDirection(Direction::Input);
-    mBuilder.setAudioApi(AudioApi::OpenSLES);
 	openStream();
+	ASSERT_EQ((int)AudioApi::OpenSLES, (int)mStream->getAudioApi());
 	ASSERT_EQ((int)mStream->getPerformanceMode(), (int)PerformanceMode::None);
     closeStream();
 }
+
+TEST_F(StreamOpen, ForOpenSlesIllegalFormatRejectedOutput) {
+    mBuilder.setAudioApi(AudioApi::OpenSLES);
+    mBuilder.setPerformanceMode(PerformanceMode::LowLatency);
+    mBuilder.setFormat(static_cast<AudioFormat>(666));
+    Result r = mBuilder.openStream(&mStream);
+    EXPECT_NE(r, Result::OK) << "Should not open stream " << convertToText(r);
```

Comment:
Wouldn't it be better to check for `oboe::Result::ErrorInvalidFormat`?

Reply:
It is currently returning ErrorInternal.  

I think we should add some more general parameter checking to AudioStreamBuilder. That will get called in the open() and return a more suitable error message.